### PR TITLE
Various fixes / cleanup to SerialPort test suite

### PR DESCRIPTION
**Pull request by willdean**
This contains a number fairly minor changes, which are probably in three categories - cosmetic cleanup, fixes for problems introduced by me in the migration and fixes for problems in the original test suite which have only worked by luck in the past (changes like ReadLine.cs are in this last category)

If the project would prefer me to supply more, smaller, PRs I can do that - I am trying to strike some reasonable balance.

It's not worth calling-out comment typos on review at this stage, there are thousands.

### CODE REVIEW

File: src/System.IO.Ports/tests/Legacy/SerialPort/DosDevices.cs [hunk 1]
```diff
@@ -161,5 +138,5 @@ private static char[] CallQueryDosDevice(string name, out int dataSize)
     public const int ERROR_MORE_DATA = 234;
 
     [DllImport("Kernel32.dll", SetLastError = true, CharSet = CharSet.Auto)]
-    internal static extern int QueryDosDevice(string lpDeviceName, char[] lpTargetPath, int ucchMax);
+    static extern int QueryDosDevice(string lpDeviceName, char[] lpTargetPath, int ucchMax);
```
Comment (JeremyKuhne) [inline src/System.IO.Ports/tests/Legacy/SerialPort/DosDevices.cs]: Should be explicit in all cases per our style guidelines

File: src/System.IO.Ports/tests/Legacy/SerialPort/Event_Generic.cs [hunk 2]
```diff
@@ -152,7 +152,7 @@ public void EventHandler_ThrowsException()
         }
     }
 
-    [ConditionalFact(nameof(HasNullModem))]
+    [ConditionalFact(nameof(HasNullModem), Skip="Always hangs")]
```
Comment (JeremyKuhne) [inline src/System.IO.Ports/tests/Legacy/SerialPort/Event_Generic.cs]: Should have an associated issue, we generally don't use Skip directly. The only exception would be a test that you're expected to manually run, which should have a detailed comment explaining that.

**Comment by willdean** [comment]
@JeremyKuhne Thanks for the review - I've rebased the branch, addressed those two nits,  and force-pushed it - hopefully that hasn't sent everything spinning off into the weeds - I'll do a new PR if it has.

**Comment by willdean** [comment]
Actually it appears I haven't rebased it at all, sigh.  Leave it with me for a minute

**Comment by JeremyKuhne** [comment]
@dotnet-bot test this please (rerun to ensure stability)

**Comment by JeremyKuhne** [comment]
@dotnet-bot test this please (rerun to ensure stability)

**Comment by willdean** [comment]
@JeremyKuhne Is dotnet-bot a person or an automaton?  Is the protocol documented anywhere?

**Comment by JeremyKuhne** [comment]
@willdean it is a bot, you can ask it to kick a specific run or all of them. Send it "help" to get the full list.

**Comment by JeremyKuhne** [comment]
@willdean I fixed the conflict as it was just the comment spacing

**Comment by willdean** [comment]
Thanks!

**Comment by willdean** [comment]
@dotnet-bot test this please (rerun to ensure stability)

**Comment by willdean** [comment]
@dotnet-bot help

**Pull request merged**